optuna/trial.py

```diff
@@ -732,14 +732,21 @@ def objective(trial):
                 Lower endpoint of the range of suggested values. ``low`` is included in the range.
             high:
                 Upper endpoint of the range of suggested values. ``high`` is included in the range.
+            log:
+                A flag to sample the value from the log domain or not.
+                If ``log`` is true, the value is sampled from the range in the log domain.
+                Otherwise, the value is sampled from the range in the linear domain.
             step:
                 A step of spacing between values.
 
         Returns:
             A suggested integer value.
         """
 
-        distribution = distributions.IntUniformDistribution(low=low, high=high, step=step)
+        if log:
+            distribution = distributions.IntLogUniformDistribution(low=low, high=high, step=step)
+        else:
+            distribution = distributions.IntUniformDistribution(low=low, high=high, step=step)
```

Comment:
To avoid mypy error, @himkt and I come up with two solutions:

**1: Annotate `IntUniformDistribution`**

```python
if log:
    distribution: IntUniformDistribution = distributions.IntLogUniformDistribution(low=low, high=high, step=step)
else:
    distribution = distributions.IntUniformDistribution(low=low, high=high, step=step)
```

**2: Change the order of inside**
```python
if not log:
    distribution = distributions.IntUniformDistribution(low=low, high=high, step=step)
else:
    distribution = distributions.IntLogUniformDistribution(low=low, high=high, step=step)
```

But we aren't sure these ways are fine...